src/dotnet-new3/NewtonsoftJsonDocumentObjectModel.cs

```diff
@@ -160,11 +187,24 @@ public IJsonArray RemoveAt(int index)
         private class JValueAdapter : JTokenAdapter, IJsonValue
         {
             public JValueAdapter(JToken token, IJsonDocumentObjectModelFactory factory)
-                : base (token, factory)
+                : base(token, factory)
             {
             }
 
-            public object Value => Token.Value<object>();
+            public object Value
+            {
+                get
+                {
+                    object val = ((JValue)Token).Value;
+                    switch (Token.Type)
+                    {
+                        case JTokenType.Integer:
+                            return (double)(long)val;
```

Comment:
Why cast twice?

Reply:
JSON.NET stores integer types as `long` so we need to unbox as the appropriate type before storing as `double`